shell/common/rasterizer.cc

```diff
@@ -533,15 +533,41 @@ RasterStatus Rasterizer::DrawToSurfaceUnsafe(
       external_view_embedder_.get(),  // external view embedder
       root_surface_transformation,    // root surface transformation
       true,                           // instrumentation enabled
-      frame->supports_readback(),     // surface supports pixel reads
-      raster_thread_merger_           // thread merger
+      frame->framebuffer_info()
+          .supports_readback,  // surface supports pixel reads
+      raster_thread_merger_    // thread merger
   );
   if (compositor_frame) {
-    RasterStatus raster_status = compositor_frame->Raster(layer_tree, false);
+    // Disable partial repaint if external_view_embedder_ SubmitFrame is
+    // involved - ExternalViewEmbedder unconditionally clears the entire
+    // surface and also partial repaint with platform view present is something
+    // that still need to be figured out.
+    bool disable_partial_repaint =
+        external_view_embedder_ &&
+        (!raster_thread_merger_ || raster_thread_merger_->IsMerged());
+
+    FrameDamage damage;
```

Comment:
I think making `FrameDamage` its own class and having the following methods will be useful:

* ctor which takes in a `prev_layer_tree`
* `GetFrameDamage`
* `GetBufferDamage`
* `AddAdditionalDamage`

Reply:
The constructor part would be bit cumbersome because `prev_layer_tree` is null in case damage is not being computed. Other than that done [here](https://github.com/flutter/engine/pull/28801/commits/9f4ebe032056bc6032e1c6c98d446719b9de2a6d).

Reply:
👍🏽